File: client/src/models/Restaurant.js
import { APIItem } from "./APIItem.js"
import { Profile } from "./Profile.js"

export class Restaurant extends APIItem {
  constructor(data) {
    super(data)
    this.creatorId = data.creatorId
    this.description = data.description
    this.imgUrl = data.imgUrl
    this.isShutdown = data.isShutdown
    this.name = data.name
    this.reportCount = data.reportCount
    this.visits = data.visits
    this.owner = new Profile(data.owner)
  }
}

File: server/Models/Restaurant.cs
using System.ComponentModel.DataAnnotations;

namespace help_reports.Models;

public class Restaurant : RepoItem<int>
{
  public string Name { get; set; }
  [Url] public string ImgUrl { get; set; }
  public string Description { get; set; }
  // TODO make sure this is actually implemented in the repository
  public int Visits { get; set; }
  public bool? IsShutdown { get; set; }
  public string CreatorId { get; set; }
  public Profile Owner { get; set; }
  // TODO make sure this is actually implemented in the repository
  public int ReportCount { get; set; }
}